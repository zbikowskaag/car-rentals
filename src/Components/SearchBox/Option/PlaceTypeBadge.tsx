import React from 'react';

interface PlaceTypeBadgeProps {
    placeType: string;
}

const PlaceTypeBadge = (props: PlaceTypeBadgeProps) => {
    let text: string;
    let className: string;

    switch (props.placeType) {
        case 'A':
            text = 'Airport';
            className = 'orange';
            break;
        case 'C':
            text = 'City';
            className = 'blue';
            break;
        case 'T':
            text = 'Station';
            className = 'blue';
            break;
        case 'D':
            text = 'Station';
            className = 'blue';
            break;
        default:
            text = 'Other';
            className = 'blue';
    }

    return (
        <div className="placeTypeBadge flex itemsCenter">
            <span className={`placeTypeBadgeText placeTypeBadgeText-${className}`}>{text}</span>
        </div>
    );
};

export default PlaceTypeBadge;
