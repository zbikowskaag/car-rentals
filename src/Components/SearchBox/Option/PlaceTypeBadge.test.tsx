import React from 'react';
import { render } from '@testing-library/react';
import PlaceTypeBadge from './PlaceTypeBadge';

describe('<PlaceTypeBadge>', () => {
    function setup(params: { placeType: string }) {
        return render(<PlaceTypeBadge placeType={params.placeType} />);
    }

    it('renders the blue badge', () => {
        const target = setup({
            placeType: 'C',
        });

        expect(target).toMatchSnapshot();
    });

    it('renders the orange badge', () => {
        const target = setup({
            placeType: 'A',
        });

        expect(target).toMatchSnapshot();
    });
});
