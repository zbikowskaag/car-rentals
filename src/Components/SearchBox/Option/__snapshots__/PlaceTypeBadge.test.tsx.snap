// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlaceTypeBadge> renders the blue badge 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="placeTypeBadge flex itemsCenter"
      >
        <span
          class="placeTypeBadgeText placeTypeBadgeText-blue"
        >
          City
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="placeTypeBadge flex itemsCenter"
    >
      <span
        class="placeTypeBadgeText placeTypeBadgeText-blue"
      >
        City
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<PlaceTypeBadge> renders the orange badge 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="placeTypeBadge flex itemsCenter"
      >
        <span
          class="placeTypeBadgeText placeTypeBadgeText-orange"
        >
          Airport
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="placeTypeBadge flex itemsCenter"
    >
      <span
        class="placeTypeBadgeText placeTypeBadgeText-orange"
      >
        Airport
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
